lcls-twincat-sample-delivery/Library/POUs/VICI/FB_4PViciValve.TcPOU

```diff
@@ -0,0 +1,68 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4024.12">
+  <POU Name="FB_4PViciValve" Id="{867cd0ab-cbb3-428d-92e9-f85cb5a24957}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_4PViciValve
+VAR_INPUT
+    iAddress : INT;
+END_VAR
+VAR_OUTPUT
+END_VAR
+VAR
+
+    // Re-Use Vici Driver Code for 12 port Value
+    fbViciDriver  : FB_ViciDriver;
+    fbViciSerial  : FB_SerialCommWrapper;
+    stViciSerial  : ST_SerialComm;
+    stViciVlvCtrl : ST_ViciControl;
+    stViciVlvStatus : ST_ViciStatus;
+    //
+    iNewPos: INT;
+    iMaped : INT;
+
+
+    {attribute 'pytmc' := '
+        pv: ReqPos
+        io: io
+    '}
+    iVici1ReqPos : INT;
+
+    {attribute 'pytmc' := '
+        pv: SampleNames
+        io: io
+     '}
+     stSampleName : ST_4PViciValveSampleNames;
+
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[// Re-using 12 Port Code, Need to map 1-4 to vlaues that 12 port Commands
+//map 1-4 to 9 - 12;
+
+iNewPos := LIMIT(1, iVici1ReqPos, 4);
+iMaped := 12 + ((iNewPos - 1)*(9 - 12) / (4 - 1));
```

Comment:
Can this be rewritten more straightforwardly for maintainability purposes? It looks like the effect here is to map 1 -> 12, 2 -> 11, 3 -> 10, and 4 -> 9?